orchestrator/gravity_utils/src/types/ethereum_events.rs

```diff
@@ -818,6 +759,106 @@ impl LogicCallExecutedEvent {
     }
 }
 
+fn parse_string(data: &[u8], arg_index: usize) -> Result<String, GravityError> {
+    // fetching the string from the first 32 bytes
+    let offset_start = arg_index * 32;
+    let offset_end = offset_start + 32;
+    let offset = Uint256::from_bytes_be(&data[offset_start..offset_end]).unwrap();
+
+    // parse start and end of the string length
+    let len_start_index: usize = offset.to_string().parse().unwrap();
```

Comment:
there is a [try_resize_to_usize](https://docs.rs/u64_array_bigints/0.3.1/u64_array_bigints/struct.U256.html#method.try_resize_to_usize) function, so you don't have to serialize and immediately deserialize. Also the cases where it doesn't fit in a `usize` or the index is out of bounds should be handled